runtimes/peregrine/src/kilt/mod.rs

```diff
@@ -112,9 +114,21 @@ parameter_types! {
 	pub const NativeAsset: Location = Junctions::Here.into_location();
 }
 
+pub struct MetadataProvider;
```

Comment:
This looks to me an overkill. I would also not put this stuff in `constants`, but would declare them in each runtime. So basically I would introduce two more consts next to where you define `CURRENCY_SYMBOL`. As for solving the problem with `NativeAndForeignAssets`, I would simply move that inside Peregrine runtime, and define a different one for Spiritnet later on.

Reply:
I thought about this, and actually I think it's ok to leave it in `runtime_common`. Please use types instead of tuples for the return type, and perhaps quickly check if there's already a trait that could be used, before declaring our own.

Reply:
I couldn't find a trait for the native assets. There is one for `fungibles` which is not perfectly suited. 

Let me know what you think: [77ce5ce](https://github.com/KILTprotocol/kilt-node/pull/818/commits/77ce5ce2fa74fe1c89ceb62f31fd42c4b9fe1895)